test/retrieval_tests.js

```diff
@@ -0,0 +1,47 @@
+import test, { describe } from 'node:test'
+import assert from 'node:assert/strict'
+import { checkRetrieval, retrieveFile } from '../lib/retrieval.js'
+import fs from 'fs/promises'
+import { createHash } from 'crypto'
+import { fileURLToPath } from 'url'
+import path from 'path'
+
+describe('checkRetrieval', () => {
+  const sampleCid =
+    'baga6ea4seaqkzso6gijktpl22dxarxq25iynurceicxpst35yjrcp72uq3ziwpi'
+  const baseUrl = 'yablu.net'
+
+  test('should download PDF file from live URL', async () => {
+    const result = await checkRetrieval(baseUrl, sampleCid)
+    assert.equal(result, true, 'checkRetrieval should return true on success')
+  })
+
+  test('should return file with expected hash', async () => {
+    const downloadedData = await retrieveFile(baseUrl, sampleCid)
+    assert.ok(
+      downloadedData,
+      'Downloaded data should not be null or undefined'
+    )
+
+    // Check the hash of the testData against the fetched data
+    const __filename = fileURLToPath(import.meta.url)
+    const __dirname = path.dirname(__filename)
+    const testFilePath = path.join(__dirname, 'testData.pdf')
+    const testData = await fs.readFile(testFilePath)
+
+    const expectedHash = createHash('sha256').update(testData).digest('hex')
```

Comment:
We can save `expectedHash` as a constant rather then computing it for every time we run this test. This could also enable us to remove the `testData.pdf` file from the repository.